### DISCUSSION TITLE
All new bugs found in your app

### DISCUSSION THREAD

**Discussion by stan6** [SOEN345-WINTER2024/discussion#12] [Q&A]
Put the bug here. Read the https://github.com/orgs/SOEN345-WINTER2024/discussions/3 for example bug report

**Comment by ramy-att** [comment]
# RedReader Bug:
Ramy Attalla - 40205642

- Device Info: Used Pixel 4A, android version 2024010900
- Version tested: V1.23.1 stable
- Stacktrace: Bug does not cause a crash
- Duplicate of https://github.com/QuantumBadger/RedReader/issues/1143

## How to reproduce:
1. Launch the app
2. Open any subreddit
3. Open any picture
4. Swipe screen left to show the sidebar
5. The screen goes black and shows scrollbars

## Actual behavior:
![image](https://github.com/SOEN345-WINTER2024/discussion/assets/83047251/3848584e-b44d-4b37-a071-50937635739d)

## Expected behavior:
Sidebar shows up with all the options

**Reply by stan6** [reply]
Hi @ramy-att . Thanks for the bug report. However, please note that we only give bonus point for new bugs so duplicated bugs are not counted. We have this rule to avoid students reporting all the old bugs here.

**Comment by TarekElalfi** [comment]
###  SmartTube Bug:
Tarek Elalfi 40197527

- Stacktrace: The Bug does not make the app crash
- Device Information: Used Android 8.1 (Andriod TV)
- Version tested: Version 21.54 Stable
- Patch: Could not issue a pull request
- Duplicate: This bug is new and not duplicated

**How to reproduce:**
1. Launch the app
2. select a Video from any row other than the topmost 

**Actual behaviour:**

![1](https://github.com/SOEN345-WINTER2024/discussion/assets/98570182/8738fe14-db5a-4fb8-8ab4-b8435d0a7919)


**Expected behaviour:**
The dashboard (Search and profile Icons should Appear) as shown below:

![2](https://github.com/SOEN345-WINTER2024/discussion/assets/98570182/35f2177c-b4e3-41e5-89d5-aeb170d8db6a)

**Reply by stan6** [reply]
Hi, @TarekElalfi thanks for the bug report. Can you include the link for the GitHub issue in the open-source app where you have reported the bug? It is important for us to keep track and make sure that it is not an old/duplicated bug.

**Reply by stan6** [reply]
This is the link for SmartTube Issue discovered:

Bug Profile and search icon dispears · Issue #3383 · yuliskov/SmartTube (github.com)<[yuliskov/SmartTube#3383](https://github.com/yuliskov/SmartTube/issues/3383)>

**Comment by KA-devl** [comment]
### Unitto android app Bug:
Khadiri Anes 40159080

Stacktrace: No stack trace (UI unexpected behaviour/Bug)
Device Information: Android (Pixel 3A) or any other android device
Version Name: Quick Silver
Version Code: 36
Duplicate: This bug have not been posted before

### How to reproduce

1 - Launch the Unitto app
2 - Navigate to the "Hamburger" menu and click on BMI calculator
3 - Input any non negative numerical value in the input fields

### Actual behaviour :
The input field only validates negative, and character values. However, it does not restrict the user from entering large & impossible values which end up leading to an overflow of numbers in the GUI & potentially crashing the application.
![Screenshot 2024-03-19 232441](https://github.com/SOEN345-WINTER2024/discussion/assets/83729449/694e0577-1913-4053-9956-cad701d79889)

### Expected behaviour :
The input field should limit the Height and the weight to a reasonable value. As a reference, it can be limited to the height of the tallest human in the world (9feet) and biggest human in the world (1000lb)

**Reply by stan6** [reply]
Hi, @KA-devl thanks for the bug report. Can you include the link for the GitHub issue in the open-source app where you have reported the bug? It is important for us to keep track and make sure that it is not an old/duplicated bug.

**Comment by ramy-att** [comment]
# RedReader Bug:

Ramy Attalla - 40205642

Device Info: Used Pixel 4A, android version 2024010900
Version tested: V1.23.1 stable
Stacktrace: Bug does not cause a crash
Duplicate: None

## How to reproduce:

1. Launch app
2. Open any subreddit and go to a post without links
3. Displays (self.subredditName)

## Actual Behavior:

![Screenshot_20240321-225137 (1)](https://github.com/SOEN345-WINTER2024/discussion/assets/83047251/82dcc766-48c4-4fed-9d7d-d39e41b4b81c)


## Expected Behavior:

1. If there is a link: display the link as a source link
2. If there is no link: should not display anything

![Screenshot_20240321-230251](https://github.com/SOEN345-WINTER2024/discussion/assets/83047251/7a261a8b-f9d6-4666-8d2c-4cd5ed302c38)

![image](https://github.com/SOEN345-WINTER2024/discussion/assets/83047251/02cbaf82-3383-471f-a70a-d977c9066ffb)

**Reply by ramy-att** [reply]
Hi @stan6 found another unfound bug!

**Reply by stan6** [reply]
Hi, thanks for the bug report. Can you include the link for the GitHub issue in the open-source app where you have reported the bug? It is important for us to keep track and make sure that it is not an old/duplicated bug.

**Reply by ramy-att** [reply]
https://github.com/QuantumBadger/RedReader/issues/1180

**Comment by Itek01** [comment]
# Title: Creation of Null Decks

## Description
The user can create infinitely many null decks when creating flash quizzes cards. This bug does not lead to crashes but can produce a very messy interface.

![image](https://github.com/SOEN345-WINTER2024/discussion/assets/79113473/f8e81731-3eb3-43fa-bcc3-57a32d469cc8)


# Steps to Reproduce

1. Open the "A Flash Deck" app.
2. Create a new Deck and give it any non-blank name.
3. Save The Deck
4. Edit this Deck
5. You can now give it a blank name (Not supposed to)

# Expected Behavior
![image](https://github.com/SOEN345-WINTER2024/discussion/assets/79113473/356fd490-6cd4-4a0e-91ec-f246f7874a94)

# Actual Behavior
![image](https://github.com/SOEN345-WINTER2024/discussion/assets/79113473/a43d6f77-9c1a-4d19-8a51-cf8dfd13008b)
## Device Information
**Device Model:** Google Pixel 6
**Android Version:** Android 8.1
**App Version:** 1.7.6

### Stacktrace 
```
(No Crash for this bug)
```
### Suggested Patch
A validity check already exists when adding a deck or card. Simply using the same one for editing a card or deck would solve this issue.
### Duplicate Check
I have searched the existing issues and confirm that this bug has not been reported before.

**Reply by stan6** [reply]
Hi, @Itek01  thanks for the bug report. Can you include the link for the GitHub issue in the open-source app where you have reported the bug? It is important for us to keep track and make sure that it is not an old/duplicated bug.

**Reply by Itek01** [reply]
Here is the link:

https://github.com/rh-id/a-flash-deck/issues/22

**Comment by mario-elshaer** [comment]
# Bug Report: Kolab Notes Bug

## Description
If the user decides to draw and beginnings by drawing lines consecutively, some lines are only registered as dots.

## Environment
- **Device Info**: Pixel 3a API 29, android version 2024010900
- **Kolab Notes Version**: 4.21
- **Android Version**: 231.9392.1.2311.1133
- **Stacktrace**: "Bug does not cause a crash"
- **Is it a Duplicate?**: No

## Steps to Reproduce
1. Launch App
2. Open a Notebook
3. Click on the note section
4. Click the third icon (squiggly)
5. Start drawing lines and the bug will eventually show

## Expected Behavior
You see the lines you drew properly

## Actual Behavior
You see dots someplace instead of lines

## Additional Context
<img width="405" alt="Screenshot 2024-03-22 at 6 07 18 PM" src="https://github.com/SOEN345-WINTER2024/discussion/assets/103081449/70ebfedd-ddda-4c1f-8f9b-b515cd4a2ebf">

**Reply by stan6** [reply]
Hi, thanks for the bug report. Can you include the link for the GitHub issue in the open-source app where you have reported the bug? It is important for us to keep track and make sure that it is not an old/duplicated bug.

**Reply by mario-elshaer** [reply]
https://github.com/konradrenner/kolabnotes-android/issues/220

**Comment by yep-pushian** [comment]
**Yeprem Antranik (40204291)**
## Bug Report: Kolab Notes Bug

### Description
In the app, selecting text and applying a header style mistakenly formats the entire line instead of just the highlighted section.

### Environment
- Device Info: Pixel 3a API 29, android version 2024010900
- Kolab Notes Version: 4.21
- Android Version: 231.9392.1.2311.1133
- Stacktrace: "Bug does not cause a crash"
- Is it a Duplicate?: No

### Steps to Reproduce
1. Open App
2. Create a Notebook
3. Click on the note section
4. Write a text
5. Select any part of the text 
6. Apply any heading style

### Expected Behaviour 
Heading style only applied to the selected text 

### Actual Behaviour
Heading style applies to entire line of written text

### Additional Context

<img width="234" alt="Screenshot 2024-03-22 at 6 25 28 PM" src="https://github.com/SOEN345-WINTER2024/discussion/assets/102338675/925f47cb-1e9e-4f84-840c-a09583aaf6ff">

**Reply by stan6** [reply]
Hi, thanks for the bug report. Can you include the link for the GitHub issue in the open-source app where you have reported the bug? It is important for us to keep track and make sure that it is not an old/duplicated bug.

**Reply by yep-pushian** [reply]
https://github.com/konradrenner/kolabnotes-android/issues/221

**Comment by jcjc1233** [comment]
## Jean-Claude Abou-Elias (ID 40086851)

## Bug Report: android-money-manager-ex

### Description
In the app, when I want to show a chart for a specific currency the screen will show blank with no error message at all.

### Environment
- Device Info: Samsung S22 Ultra 
- android-money-manager-ex Version: 2024.02.15 (1019)
- Android Version: 14
- Stacktrace: Bug does not cause a crash
- Is it a Duplicate?: No

### Steps to Reproduce
1. Open App
2. Click “hamburger” icon
3. Click on Entities
4. Click on Currencies
5. Choose currency type (e.g., From which country)
6. Click on show currency chart
7. Screen will go blank 

### Expected Behaviour
The app is supposed to show the complete chart of the selected currency.

### Actual Behaviour
![WhatsApp Image 2024-03-22 at 21 07 19_071f2941](https://github.com/SOEN345-WINTER2024/discussion/assets/98419894/a0d420eb-2796-409c-841f-aa6bf3dc4501)

### After clicking show chart this is the blank screen
![WhatsApp Image 2024-03-22 at 21 07 19_6e605848](https://github.com/SOEN345-WINTER2024/discussion/assets/98419894/f2dba96f-e927-4da8-9e34-e40bbd48ab80)

**Reply by stan6** [reply]
Hi, @jcjc1233 thanks for the bug report. Can you include the link for the GitHub issue in the open-source app where you have reported the bug? It is important for us to keep track and make sure that it is not an old/duplicated bug.

**Reply by jcjc1233** [reply]
https://github.com/moneymanagerex/android-money-manager-ex/issues/1666

**Comment by NicholasWahome** [comment]
Nicholas Werugia 40131956
Search option bar disappearance

Description: 
New bug discovered, when you navigate to the search options button at the top right of the search page and press on it. The lower navigation bar disappears.
Inputs: Not text, but must click button on lowerbar of page and option button
Stacktrace: Does not lead to a crash
Device Information: PIXEL 6 API 34 (used emulator)
Version of app: 4.3.0
Duplicate: No (New bug)
Steps to reproduce: 
1. Navigate to search page by pressing on the search button button at the bottom of the lower navigation bar. 
2. Press on the search bar optionn button at the top right
3. The lower navigation bar will then disappear. 

Expected Behaviour:
<img width="322" alt="Screenshot 2024-03-22 at 10 58 14 PM" src="https://github.com/SOEN345-WINTER2024/discussion/assets/87381233/7d89fb73-a551-47d6-ab57-a739b2aa3316">

Actuval Behvaiour:
<img width="322" alt="Screenshot 2024-03-22 at 10 58 22 PM" src="https://github.com/SOEN345-WINTER2024/discussion/assets/87381233/809e73af-504b-4498-9742-d40f10ce2e89">

Patch issue: 
<img width="651" alt="Screenshot 2024-03-22 at 11 20 19 PM" src="https://github.com/SOEN345-WINTER2024/discussion/assets/87381233/5978ac2d-9061-4bc0-bfc7-d7a5931dd95e">

**Reply by stan6** [reply]
Hi, @NicholasWahome thanks for the bug report. Can you include the link for the GitHub issue in the open-source app where you have reported the bug? It is important for us to keep track and make sure that it is not an old/duplicated bug.

**Reply by NicholasWahome** [reply]
https://github.com/recloudstream/cloudstream/issues/1068

**Comment by jesseythach** [comment]
# Images-to-PDF App Bug Report:

## How to reproduce:
1. Launch the app.
2. On the main page, select "Images to PDF".
3. Click on "SELECT IMAGES", select images from the device, and click on "Apply".
4. Select either "YES" or "NO" for image cropping.
5. Rapidly click on "CREATE PDF".

## Actual Behavior
Once we get to this behavior (page), the app stops responding. Clicking on any buttons on the screen doesn't do anything. The user must quit the app and relaunch it again.
<img src="https://github.com/SOEN345-WINTER2024/discussion/assets/94651491/5ea57709-6ca3-4f0e-b2ec-48e435fa0588" width="500">

## Expected Behavior
The user should be on this page where they can rename the pdf file.
<img src="https://github.com/SOEN345-WINTER2024/discussion/assets/94651491/d01c7cd2-2e32-4eba-b3bf-fe42d91a7a32" width="500">

App version: 9.0.1
Android version: 14

**Reply by stan6** [reply]
Hi, thanks for the bug report. Can you include the link for the GitHub issue in the open-source app where you have reported the bug? It is important for us to keep track and make sure that it is not an old/duplicated bug.

**Reply by jesseythach** [reply]
Here's the link to the issue of the bug report: https://github.com/Swati4star/Images-to-PDF/issues/1160

**Comment by justncheng** [comment]
# 2048 Game bug

Justin Cheng 40210279
Device Model: Pixel 3a API 34
Android Version: Android 14.0
App version: v1.2.8-fdroid

Github Issue: https://github.com/andstatus/game2048/issues/21

### Description: 

During testing using Monkey, from the Android Debug Bridge (adb) suite, a bug was found where the getCharacter method gets called in the KorgwSurfaceView.onKeyMultiple method. I came across this error multiple times, as Monkey was going through and inputting random inputs. I know that the getCharacter call is causing the issue, as it is returning a null value, however, I am not sure how it affects the app itself, as I was not able to find where the issue resides in the app. For this reason, I did not include an actual and expected behaviour, simply because I am not sure what this bug affects.

### Where is the issue: 

The issue is contained in line 234 in the KorgwSurfaceView class in the onKeyMultiple method. The stack trace indicates that there is an issue with the getCharacter(..) method, which returns a nullPointerException.

**Reply by stan6** [reply]
hi, @justncheng, usually bug reports would not require developer to install any software (monkey). You need to modify your submitted bug report to remove all the steps about monkey. 1. You need to run monkey by yourself and monitor what are the UI actions (clicking which button) cause the crash. Even if you couldn't determine the actions causing the crash , you should not ask developer to run monkey (because the developer will simply ignore your bug report or even reject your bug report), 2. your submitted report to 2048 should include the information that you have included here " know that the getCharacter call is causing the issue, as it is returning a null value, however, I am not sure how it affects the app itself, as I was not able to find where the issue resides in the app" so that developer will know that there is a NullPointerException.

**Reply by justncheng** [reply]
Thank you, I made the necessary changes to the bug submission

**Comment by vani2525** [comment]
Vanisha Patel 40242554

[Chicago city public Transport network not supported #943](https://github.com/grote/Transportr/issues/943)

**Describe the bug**
The website says that the application supports Chicago city's RTA network. When I use the application to get directions in Chicago, the app is not suggesting locations or directions.

**To Reproduce**
Steps to reproduce the behavior:
1. Set the country to United States
2. Set transport network to RTA
3. Search a destination (there are no suggested destinations)


![Screenshot_20240415_185804_Transportr 2](https://github.com/grote/Transportr/assets/133712441/b9b6a051-eb4d-40f9-b5d0-cfbadc2f1d2a)



**Expected behavior**
There should be suggested destinations and give directions to get to my destination.


**Versions (please complete the following information):**
 - **Transportr Version:** [2.21]
 - **Device:** [Samsung Galaxy A51]
 - **Android Version:** [13]

**Comment by SamyArab** [comment]
Samy Arab - 40209595
SmartTube (https://github.com/yuliskov/SmartTube) Bug: 
Link to the issue: https://github.com/yuliskov/SmartTube/issues/3434

**Describe the bug**
An initial click on the accounts button will open up the Accounts section on the right, but clicking it again doesn't close it.

**To Reproduce**
Steps to reproduce the behaviour:
Open the application
Click on Accounts
Wait for it to open on the right side
Click again on Accounts
Observe error

**Expected behaviour**
Accounts should close on pressing the button again

**Screenshots/videos**
<img width="493" alt="Screenshot 2024-04-15 at 22 09 06" src="https://github.com/yuliskov/SmartTube/assets/61292039/ee611cfd-48bf-4c8f-8d8e-8e77352207cb">


Smartphone (please complete the following information):

Device: Android TV (1080p)
OS: Android 8.1
App version: latest
Language [e.g English]
